### PR TITLE
Draft: Merge seperate animation files via vite plugin

### PR DESCRIPTION
**Pull request by oWave**
- During build: Bundles all json files in the `animations` folder into one `animations.json`
- During serve (dev): Intercepts requests to `dist/animations.json` and responds with a generated bundle.
  - `handleHotUpdate` still works. Any changes to a json file in `animations` refreshes the entire bundle

TODOs (maybe)
- Conflict detection if two files have the same keys?

### CODE REVIEW

**Comment by MrVauxs** [comment]
> TODOs (maybe)
> 
> * Conflict detection if two files have the same keys?

My idea is to create an index of duplicate keys during merging (just let JS resolve it by itself) and send it to the client to be shown through a `ui.notifications.warn`

**Comment by oWave** [comment]
Alright, done. Also snuck some lines in there that create empty `dist/pf2e-graphics.{js,css}` files during dev if they don't exist yet, so foundry stops complaining

File: vite.config.ts [hunk 1]
```diff
@@ -114,19 +115,92 @@ export default defineConfig(({ command: _buildOrServe }) => ({
 			},
 		},
 		{
-			name: 'update-animations-json',
-			handleHotUpdate({ file, server, read: _read }) {
-				if (file.endsWith('animations.json')) {
+			name: 'create-dist-files',
+			apply: 'serve',
+			buildStart() {
+				const files = [...moduleJSON.esmodules, moduleJSON.styles].flat()
```
Comment (MrVauxs) [inline vite.config.ts:121]: Can we be consistent here, use `...` on both or remove the spreader and just use `flat`

File: vite.config.ts [hunk 2]
```diff
@@ -114,19 +115,92 @@ export default defineConfig(({ command: _buildOrServe }) => ({
...
+			},
+		},
+		mergeAnimationsPlugin(),
+	],
+}))
+
+function mergeAnimationsPlugin(): PluginOption {
+	const mergeAnimations = () => {
+		let animations = {}
+		const keys = new Set<string>()
```
Comment (MrVauxs) [inline vite.config.ts:134]: Is this necessary? Why not compare the newly added keys directly to those inside the `animations` object?

**Review by MrVauxs — CHANGES_REQUESTED** [review]

**Comment by oWave** [comment]
I forgot the in operator exists.
Side note: Setting `lib` in `tsconfig` might be nice so a few newer JS features are available (`Object.hasOwn`, `cause` for Errors)

**Pull request merged**